service/tbc/crawler.go

```diff
@@ -72,6 +72,36 @@ var (
 	}
 )
 
+type NotLinear string
+
+func (en NotLinear) Error() string {
+	return string(en)
+}
+
+func (nl NotLinear) Is(target error) bool {
+	_, ok := target.(NotLinear)
+	return ok
+}
```

Comment:
I think this type should be called `NotLinearError`, as it implements `error`.
Additionally, the receiver name is different in each function - could it be changed to one name for both?

```suggestion
type NotLinearError string

func (e NotLinearError) Error() string {
	return string(e)
}

func (e NotLinearError) Is(target error) bool {
	_, ok := target.(NotLinear)
	return ok
}
```